stempy/image.cpp

```diff
@@ -386,11 +395,76 @@ void radialSumFrame(int centerX, int centerY, const uint16_t data[],
   }
 }
 
+namespace {
+
+#ifdef VTKm
+struct RadialSumWorklet : public vtkm::worklet::WorkletMapField
+{
+  vtkm::Vec<int, 2> m_center;
+  vtkm::Vec<int, 2> m_shape;
+  int m_imageNumber;
+  uint32_t m_numberOfScanPositions;
+
+  RadialSumWorklet(vtkm::Vec<int, 2> center, vtkm::Vec<int, 2> shape,
+                   int imageNumber, uint32_t numberOfScanPositions)
+    : m_center(center), m_shape(shape), m_imageNumber(imageNumber),
+      m_numberOfScanPositions(numberOfScanPositions){};
+
+  using ControlSignature = void(FieldIn, AtomicArrayInOut);
+
+  using ExecutionSignature = void(_1, _2, WorkIndex);
+
+  template <typename AtomicInterface>
+  VTKM_EXEC void operator()(const vtkm::UInt16& value,
+                            AtomicInterface& radialSum, vtkm::Id i) const
+  {
+    auto x = i % m_shape[0];
+    auto y = i / m_shape[0];
+    auto radius =
+      static_cast<int>(std::ceil(distance(x, y, m_center[0], m_center[1])));
+
+    radialSum.Add(radius * m_numberOfScanPositions + m_imageNumber, value);
+  }
+};
+
+void radialSumFrame(const vtkm::Vec<int, 2>& center,
+                    const ArrayHandleView<vtkm::UInt16>& data,
+                    const vtkm::Vec<int, 2>& shape, int imageNumber,
+                    uint32_t numberOfScanPositions,
+                    vtkm::cont::ArrayHandle<vtkm::Int64>& radialSum)
+{
+  vtkm::worklet::Invoker invoke;
+  invoke(RadialSumWorklet{ center, shape, imageNumber, numberOfScanPositions },
+         data, radialSum);
+}
+
 void radialSumFrames(int centerX, int centerY, const uint16_t data[],
-                   int frameWidth, int frameHeight,
-                   std::vector<uint32_t>& imageNumbers,
-                   uint32_t numberOfPixels,
-                   RadialSum<uint64_t>& radialSum)
+                     int frameWidth, int frameHeight,
+                     std::vector<uint32_t>& imageNumbers,
+                     uint32_t numberOfPixels, uint32_t numberOfScanPositions,
+                     vtkm::cont::ArrayHandle<vtkm::Int64>& radialSum)
+{
+  vtkm::Vec<int, 2> center = { centerX, centerY };
+  vtkm::Vec<int, 2> shape = { frameWidth, frameHeight };
+  auto dataHandle =
+    vtkm::cont::make_ArrayHandle(data, numberOfPixels * imageNumbers.size());
+  // Use view to the array already transfered
+  for (unsigned i = 0; i < imageNumbers.size(); ++i) {
+    auto offset = i * numberOfPixels;
+    auto view =
+      vtkm::cont::make_ArrayHandleView(dataHandle, offset, numberOfPixels);
+
+    radialSumFrame(center, view, shape, imageNumbers[i], numberOfScanPositions,
+                   radialSum);
+  }
+}
+
+#else
+void radialSumFrames(int centerX, int centerY, const uint16_t data[],
+                     int frameWidth, int frameHeight,
+                     std::vector<uint32_t>& imageNumbers,
+                     uint32_t numberOfPixels,
+                     uint32_t numberOfScanPositions RadialSum<uint64_t>& radialSum)
```

Comment:
Haven't tried compiling these yet, but it looks like a comma is missing here.

Reply:
Indeed, I tried compiling this with VTKm turned off. I get a compilation error here, and I think one or two others. Might be good to check them out.

Reply:
Will do

Reply:
Done